sync-core/src/main/java/com/cloudant/sync/replication/BasicPullStrategy.java

```diff
@@ -352,7 +359,12 @@ private int processOneChangesBatch(ChangesResultWrapper changeFeeds)
     public String getReplicationId() {
         HashMap<String, String> dict = new HashMap<String, String>();
         dict.put("source", this.sourceDb.getIdentifier());
-        dict.put("target", this.targetDb.getIdentifier());
+        try {
+            dict.put("target", this.targetDb.getIdentifier());
+        } catch (DatastoreException e){
+            logger.log(Level.WARNING,"Failed to get public id using default",e);
+            dict.put("target","touchdb_defaultUUID");
```

Comment:
This method should fail if you can't get the name of the target. Otherwise you're just going to end up with incorrect checkpoint docs.